src/sentry/static/sentry/app/views/issueList/sidebar.tsx

```diff
@@ -2,64 +2,78 @@ import PropTypes from 'prop-types';
 import React from 'react';
 import createReactClass from 'create-react-class';
 import isEqual from 'lodash/isEqual';
-import debounce from 'lodash/debounce';
 import map from 'lodash/map';
 import styled from '@emotion/styled';
 
 import LoadingIndicator from 'app/components/loadingIndicator';
 import {IconClose} from 'app/icons/iconClose';
-import {queryToObj, objToQuery} from 'app/utils/stream';
+import {queryToObj, objToQuery, QueryObj} from 'app/utils/stream';
 import {t} from 'app/locale';
+import {Tag, TagCollection} from 'app/types';
+import SentryTypes from 'app/sentryTypes';
 
+import {TagValueLoader} from './types';
 import IssueListTagFilter from './tagFilter';
 
-const TEXT_FILTER_DEBOUNCE_IN_MS = 300;
+type DefaultProps = {
+  tags: TagCollection;
+  query: string;
+  onQueryChange: () => void;
+};
 
-const IssueListSidebar = createReactClass({
+type Props = DefaultProps & {
+  orgId: string;
+  tagValueLoader: TagValueLoader;
+  loading?: boolean;
+};
+
+type State = {
+  queryObj: QueryObj;
+  textFilter: string;
+};
+
+const IssueListSidebar = createReactClass<Props, State>({
   displayName: 'IssueListSidebar',
 
   propTypes: {
     orgId: PropTypes.string.isRequired,
-
-    tags: PropTypes.object.isRequired,
+    tags: PropTypes.objectOf(SentryTypes.Tag).isRequired,
     query: PropTypes.string,
     onQueryChange: PropTypes.func.isRequired,
     loading: PropTypes.bool,
     tagValueLoader: PropTypes.func.isRequired,
   },
 
-  getDefaultProps() {
-    return {
-      tags: {},
-      query: '',
-      onQueryChange: function () {},
-    };
+  defaultProps: {
+    tags: {},
+    query: '',
+    onQueryChange: function () {},
   },
 
   getInitialState() {
-    const queryObj = queryToObj(this.props.query);
+    const queryObj: QueryObj = queryToObj(this.props.query);
     return {
       queryObj,
       textFilter: queryObj.__text,
     };
   },
 
-  componentWillReceiveProps(nextProps) {
+  componentWillReceiveProps(nextProps: Props) {
     // If query was updated by another source (e.g. SearchBar),
     // clobber state of sidebar with new query.
-    const query = objToQuery(this.state.queryObj);
+    const query: string = objToQuery(this.state.queryObj);
 
     if (!isEqual(nextProps.query, query)) {
-      const queryObj = queryToObj(nextProps.query);
+      const queryObj: QueryObj = queryToObj(nextProps.query);
```

Comment:
nit: probably don't need to explicitly declare the types of the variables, `query` and `queryObj` as it can be implied from the return types of the functions